psl/diagnostics/src/error.rs

```diff
@@ -287,11 +287,57 @@ impl DatamodelError {
     }
 
     pub fn new_type_not_found_error(type_name: &str, span: Span) -> DatamodelError {
-        let msg = format!(
-            "Type \"{type_name}\" is neither a built-in type, nor refers to another model, custom type, or enum."
-        );
+        // Rust built-in types and common mistakes mapping
+        const BUILT_IN_TYPES: [&str; 18] = [
+            "bool",
+            "char",
+            "i8", "i16", "i32", "i64", "i128",
+            "u8", "u16", "u32", "u64", "u128",
+            "isize", "usize",
+            "f32", "f64",
+            "str",
+            "fn"
+        ];
+    
+        const COMMON_MISTAKES: [( &str, &str ); 2] = [
+            ("integer", "i32"),  // i32 is the default integer type in Rust
+            ("character", "char")
+        ];
```

Comment:
Isn't this fundamentally the wrong list of types this feature should be about? Isn't this about Prisma scalar types?